libcaf_net/caf/net/http/arg_parser.hpp

```diff
@@ -29,7 +29,7 @@ struct builtin_arg_parser {
     if (auto err = detail::parse(str, tmp); !err)
       return tmp;
     else
-      return {};
+      return std::nullopt;
```

Comment:
I'd say `return {};` is the preferred way, because it doesn't rely on an implicit conversion. What's the warning here?

Reply:
Here is the exact warning. Same for all `return {}` locations. 
```
warning: '<anonymous>' may be used uninitialized in this function [-Wmaybe-uninitialized]
     return {};
```
I could replace it with return `std::optional{};`